x-pack/plugin/logstash/src/main/java/org/elasticsearch/xpack/logstash/action/TransportGetPipelineAction.java

```diff
@@ -169,4 +214,8 @@ private void logFailures(MultiGetResponse multiGetResponse) {
             logger.info("Could not retrieve logstash pipelines with ids: {}", ids);
         }
     }
+
+    private String toRegex(String input) {
+        return "^".concat(input.replace(WILDCARD, ".*$"));
```

Comment:
Thanks @kaisecheng for good points.
I was also aware of your 2nd point (pipeline search worked when I sent regex patter through URL query string). Plz take a look at my latest (`Wildcard regex pattern improvement`) commit so that it covers both your concerns.

My opinion for your additional following point would be "we assume users are installed standard (not customized) Logstash and Elasticsearch and the API gets the validated pipelines by [Logstash bootstrap check](https://github.com/elastic/logstash/blob/611953bbe96b4178bfbec5029e127af880ee0fa7/x-pack/lib/config_management/bootstrap_check.rb#L19)." Let me plz know if you think we should check pattern anywhere in all pipelines related ES APIs. I think it is not necessary.
```
Currently Logstash accept alphabet, number, _, - as pipeline id and only accept * as wildcard to match any character in pipeline id.
```

Reply:
I agree with you that we don't need to give pattern checking to all APIs in this enhancement. The goal is to bring wildcard to search system index. Logstash do not have upsert feature to central management. So, aligning the search part is good enough for now. It is nice to have for future to align Kibana and Elasticsearch API.